packages/compass-preferences-model/lib/renderer-ipc.js

```diff
@@ -0,0 +1,39 @@
+const ipc = require('hadron-ipc');
+
+/**
+ * API to communicate with preferences from the electron renderer process.
+ */
+const preferencesIpc = {
+  savePreferences(attributes) {
+    if (typeof ipc?.ipcRenderer?.invoke === 'function') {
+      return ipc.ipcRenderer.invoke('compass:save-preferences', attributes);
+    }
+    return;
+  },
+  getPreferences() {
+    if (typeof ipc?.ipcRenderer?.invoke === 'function') {
+      return ipc.ipcRenderer.invoke('compass:get-preferences');
+    }
+    return;
+  },
+  getConfigurableUserPreferences() {
+    if (typeof ipc?.ipcRenderer?.invoke === 'function') {
+      return ipc.ipcRenderer.invoke('compass:get-configurable-user-preferences');
+    }
+    return;
+  },
+  onPreferencesChanged2(callback) {
```

Comment:
Is this a typo?

Reply:
No.

Reply:
I am trying to mock this method for tests and trying different options to check how it is being called and where it fails. I push intermediate results for discussion. Please ignore this kind of things until pr becomes ready for review.